### PR TITLE
Add JSS citation information

### PR DESCRIPTION
**Pull request by bkamins**
This also follows the request from JSS

### CODE REVIEW

File: CITATION.bib [hunk 1]
```diff
@@ -0,0 +1,12 @@
+@article{JSSv107i04,
+ title={DataFrames.jl: Flexible and Fast Tabular Data in Julia},
+ volume={107},
+ url={https://www.jstatsoft.org/index.php/jss/article/view/v107i04},
+ doi={10.18637/jss.v107.i04},
+ abstract={&amp;lt;p&amp;gt;DataFrames.jl is a package written for and in the Julia language offering flexible and efficient handling of tabular data sets in memory. Thanks to Julia’s unique strengths, it provides an appealing set of features: Rich support for standard data processing tasks and excellent flexibility and efficiency for more advanced and non-standard operations. We present the fundamental design of the package and how it compares with implementations of data frames in other languages, its main features, performance, and possible extensions. We conclude with a practical illustration of typical data processing operations.&amp;lt;/p&amp;gt;},
```
Comment (nalimilan) [inline CITATION.bib:6]:
```suggestion
 abstract={DataFrames.jl is a package written for and in the Julia language offering flexible and efficient handling of tabular data sets in memory. Thanks to Julia’s unique strengths, it provides an appealing set of features: Rich support for standard data processing tasks and excellent flexibility and efficiency for more advanced and non-standard operations. We present the fundamental design of the package and how it compares with implementations of data frames in other languages, its main features, performance, and possible extensions. We conclude with a practical illustration of typical data processing operations.},
```

File: CITATION.bib [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ number={4},
+ journal={Journal of Statistical Software},
+ author={Bouchet-Valat, Milan and Kamiński, Bogumił},
+ year={2023},
+ pages={1–32}
```
Comment (nalimilan) [inline CITATION.bib:11]: Is `–` accepted?
```suggestion
 pages={1--32}
```

File: docs/src/index.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ running with tabular data manipulation using the DataFrames.jl package.
 For more illustrations of DataFrames.jl usage, in particular in conjunction with
 other packages you can check-out the following resources
 (they are kept up to date with the released version of DataFrames.jl):
+* [DataFrames.jl: Flexible and Fast Tabular Data in Julia](https://www.jstatsoft.org/article/view/v107i04) article published in Journal of Statistical Software
```
Comment (nalimilan) [inline docs/src/index.md:11]:
```suggestion
* [DataFrames.jl: Flexible and Fast Tabular Data in Julia](https://www.jstatsoft.org/article/view/v107i04) article published in the *Journal of Statistical Software*
```

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
I took CITATION.bib verbatim from JSS website, but your changes are OK 😄.

**Pull request merged**